java-tools/OsmAndServer/src/main/java/net/osmand/server/api/services/OsmAndMapsService.java

```diff
@@ -725,6 +725,11 @@ private RoutingContext prepareRouterContext(String routeMode, QuadRect points, R
 		Map<String, String> paramsR = new LinkedHashMap<String, String>();
 		boolean useNativeLib = DEFAULT_USE_ROUTING_NATIVE_LIB;
 		router.setUseNativeApproximation(false);
+
+		// reset before applying
+		router.USE_HH_ROUTING = false;
```

Comment:
USE_HH_ROUTING  should be default true

Reply:
USE_HH_ROUTING must be reset here before applying real API parameter which *IS* actually true by default.

Reply:
9f8dc4f0c565a20b9232ab11e89018c910865e69